README.md

```diff
@@ -369,3 +369,148 @@ This has been tested with:
+be changed at runtime via the `config-logging` `ConfigMap` which is
+[created](./config/config-logging.yaml) when deploying the Tanzu Sources for
+Knative manifests in this repository.
+
+⚠️ **Note:** These settings will affect **all adapter** deployments. Changes to
+a particular adapter deployment are currently not possible.
+
+```
+kubectl -n vmware-sources edit cm config-logging
+```
+
+An interactive editor opens. Change the settings in the JSON object under the
+`zap-logger-config` key. For example, to change the log level from `info` to
+`debug` use this configuration in the editor:
+
+```yaml
+apiVersion: v1
+data:
+  # details omitted
+  zap-logger-config: |
+    {
+      "level": "debug"
+      "development": false,
+      "outputPaths": ["stdout"],
+      "errorOutputPaths": ["stderr"],
+      "encoding": "json",
+      "encoderConfig": {
+        "timeKey": "ts",
+        "levelKey": "level",
+        "nameKey": "logger",
+        "callerKey": "caller",
+        "messageKey": "msg",
+        "stacktraceKey": "stacktrace",
+        "lineEnding": "",
+        "levelEncoder": "",
+        "timeEncoder": "iso8601",
+        "durationEncoder": "",
+        "callerEncoder": ""
+      }
+    }
+```
+
+Save and leave the interactive editor to apply the `ConfigMap` changes.
+Kubernetes will validate and confirm the changes:
+
+```
+configmap/config-logging edited
+```
+
+To verify that the `Source` adapter owners (e.g. `vsphere-source-webhook` for a
+`VSphereSource`) have noticed the desired change, inspect the log messages of
+the owner (here: `vsphere-source-webhook`) `Pod`:
+
+```
+vsphere-source-webhook-f7d8ffbc9-4xfwl vsphere-source-webhook {"level":"info","ts":"2022-03-29T12:25:20.622Z","logger":"vsphere-source-webhook","caller":"vspheresource/vsphere.go:250","msg":"update from logging ConfigMap{snip...}
+```
+
+⚠️ **Note:** To avoid unwanted disruption during event retrieval/delivery, the
+changes are **not applied** automatically to deployed adapters, i.e.
+`VSphereSource` adapter, etc.
+
+To make the changes take affect for existing adapter `Deployments`, an operator
+needs to manually scale a particular adapter `Deployment` to `--replicas 0`. The
+`Source` controller will react to this change by creating a new instance (`Pod`)
+with the desired log level changes and setting `replicas` to the default count
+(`1`) again.
+
+```
+kubectl get vspheresource
+NAME                SOURCE                     SINK                                                                              READY   REASON
+example-vc-source   https://my-vc.corp.local   http://broker-ingress.knative-eventing.svc.cluster.local/default/example-broker   True
+
+kubectl scale deployment --replicas 0 example-vc-source-deployment
+deployment.apps/example-vc-source-deployment scaled
+```
+
+⚠️ **Note:** To avoid losing events due to this (brief) downtime, use the
+[Checkpointing](#configuring-checkpoint-and-event-replay) capability.
```

Comment:
Should we be enabling this by default so that we can roll out the deployment automatically on these changes?

Reply:
When we introduced this in an earlier version, we did not want to break users and hence the default is no checkpointing. I agree with you thought, that we should provide safe defaults. Let me create an issue to enable checkpointing by default.

Reply:
https://github.com/vmware-tanzu/sources-for-knative/issues/403